source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,209 @@
+===========================================
+Connections Survive Primary Step Down Tests
+===========================================
+
+These tests can be used to verify a driver's compliance with server discovery
+and monitoring requirements with respect to handling "not master" and
+"server shutting down" error responses from the server.
+
```

Comment:
Since there are a number of sections in this file, I think it'd make sense to insert a generated TOC here with `.. contents::`. That would make it easy to see all required tests at a glance.

Reply:
Added, thank you.